src/parser/expression/constant_value_expression.cpp

```diff
@@ -250,6 +251,59 @@ std::string ConstantValueExpression::ToString() const {
   }
 }
 
+ConstantValueExpression ConstantValueExpression::FromString(const std::string &val, uint64_t type_id) {
+  const auto cve_type = static_cast<type::TypeId>(type_id);
+
+  switch (cve_type) {
+    case type::TypeId::BOOLEAN: {
+      return ConstantValueExpression(cve_type, execution::sql::BoolVal(std::stoi(val) != 0));
+    }
+    case type::TypeId::TINYINT:
+    case type::TypeId::SMALLINT:
+    case type::TypeId::INTEGER:
+    case type::TypeId::BIGINT: {
+      return ConstantValueExpression(cve_type, execution::sql::Integer(std::stoll(val)));
+    }
+    case type::TypeId::DECIMAL: {
+      return ConstantValueExpression(cve_type, execution::sql::Real(std::stod(val)));
+    }
+    case type::TypeId::TIMESTAMP: {
+      return ConstantValueExpression(cve_type,
+                                     execution::sql::TimestampVal(execution::sql::Timestamp::FromString(val)));
+    }
+    case type::TypeId::DATE: {
+      return ConstantValueExpression(cve_type, execution::sql::DateVal(execution::sql::Date::FromString(val)));
+    }
+    case type::TypeId::VARCHAR:
+    case type::TypeId::VARBINARY: {
+      auto string_val = execution::sql::ValueUtil::CreateStringVal(val);
+      return ConstantValueExpression(cve_type, string_val.first, std::move(string_val.second));
+    }
+    default:
+      UNREACHABLE("Invalid TypeId.");
+  }
+}
+
+std::string ConstantValueExpression::GetTypeAsString() const {
```

Comment:
Why is this logic specific to CVE? This doesn't seem like it belongs here. I would think `TypeUtil` with the other method to convert `TypeId` to string.

Reply:
Moved to TypeUtil in lastest push.